packages/web5/src/web5.ts

```diff
@@ -110,20 +112,25 @@ export class Web5 {
   }
 
   /**
-   * dynamically selects up to 4 dwn hosts
+   * Dynamically selects up to 2 DWN endpoints that are provided
+   * by default during the Tech Preview period.
    */
-  static async getDwnHosts() {
+  static async getTechPreviewDwnEndpoints(): Promise<string[]> {
     const response = await fetch('https://dwn.tbddev.org/.well-known/did.json');
 
+    // Return an empty array if dwn.tbddev.org is not responding.
+    if (!response.ok) { return []; }
+
     const didDoc = await response.json();
     const [ service ] = didUtils.getServices(didDoc, { id: '#dwn', type: 'DecentralizedWebNode' });
     const { nodes } = <DwnServiceEndpoint>service.serviceEndpoint;
 
     // allocate up to 2 nodes for a user.
-    const numNodesToAllocate = Math.min(Math.floor(nodes.length / 2), 2);
-    const dwnUrls = new Set([]);
+    const dwnUrls = new Set<string>();
+    let attempts = 0;
+    const numNodesToAllocate = Math.min(nodes.length, 2);
 
-    for (let i = 0; i < numNodesToAllocate; i += 1) {
+    while(dwnUrls.size < numNodesToAllocate && attempts < nodes.length) {
```

Comment:
thoughts on the following?

```typescript
// allocate up to 2 nodes for a user.
const dwnUrls = new Set<string>();
const numNodesToAllocate = Math.min(nodes.length, 2);

for (let attempts = 0; attempts < nodes.length && dwnUrls.size < numNodesToAllocate; attempts += 1) {
  const nodeIdx = getRandomInt(0, nodes.length);
  const dwnUrl = nodes[nodeIdx];

  try {
    const healthCheck = await fetch(`${dwnUrl}/health`);
    if (healthCheck.status === 200) {
      dwnUrls.add(dwnUrl);
    }
  } catch(e: unknown) {
    // Ignore healthcheck failures and try the next node.
  }
}
```

smushes `attempts`

Reply:
kinda doesn't smush `attempts` since its still there in the `for` loop 😄 
but LGTM